### PR TITLE
[DROOLS-7225] make dsl generated rule units available via quarkus CDI

### PR DESCRIPTION
**Pull request by mariofusco**
**JIRA**: https://issues.redhat.com/browse/DROOLS-7225

### CODE REVIEW

File: drools-drl-quarkus-extension/drools-drl-quarkus-ruleunit-integration-test/src/test/java/org/drools/quarkus/ruleunit/test/RuntimeDslRuleUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    @Test
+    public void testR1() {
+        SimpleDTUnit unitData = new SimpleDTUnit();
+        unitData.getAge().set( 19 );
+        unitData.getIncidents().set( false );
+
+        // CDI version:
+        RuleUnitInstance<SimpleDTUnit> unitInstance = ruleUnit.createInstance(unitData);
+        // Programmatic version:
+        // RuleUnitInstance<SimpleDTUnit> unitInstance = RuleUnitProvider.get().createRuleUnitInstance(unitData);
```
Comment (tarilabs) [inline drools-drl-quarkus-extension/drools-drl-quarkus-ruleunit-integration-test/src/test/java/org/drools/quarkus/ruleunit/test/RuntimeDslRuleUnitTest.java]:
```suggestion
```

File: drools-drl-quarkus-extension/drools-drl-quarkus-ruleunit-integration-test/src/test/java/org/drools/quarkus/ruleunit/test/RuntimeDslRuleUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    @Inject
+    RuleUnit<SimpleDTUnit> ruleUnit;
+
+    @Test
+    public void testR1() {
+        SimpleDTUnit unitData = new SimpleDTUnit();
+        unitData.getAge().set( 19 );
+        unitData.getIncidents().set( false );
+
+        // CDI version:
```
Comment (tarilabs) [inline drools-drl-quarkus-extension/drools-drl-quarkus-ruleunit-integration-test/src/test/java/org/drools/quarkus/ruleunit/test/RuntimeDslRuleUnitTest.java]:
```suggestion
```

File: drools-drl-quarkus-extension/drools-drl-quarkus-ruleunit-integration-test/src/test/java/org/drools/quarkus/ruleunit/test/RuntimeDslRuleUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        RuleUnitInstance<SimpleDTUnit> unitInstance = ruleUnit.createInstance(unitData);
+        // Programmatic version:
+        // RuleUnitInstance<SimpleDTUnit> unitInstance = RuleUnitProvider.get().createRuleUnitInstance(unitData);
+
+        unitInstance.fire();
+
+        assertThat( unitData.getBasePrice() ).hasValue( 800 );
+        System.out.println("Base price: " + unitData.getBasePrice().get());
```
Comment (tarilabs) [inline drools-drl-quarkus-extension/drools-drl-quarkus-ruleunit-integration-test/src/test/java/org/drools/quarkus/ruleunit/test/RuntimeDslRuleUnitTest.java]:
```suggestion
```

File: drools-drl-quarkus-extension/drools-drl-quarkus-util-deployment/src/main/java/org/drools/drl/quarkus/util/deployment/DroolsQuarkusResourceUtils.java [hunk 4]
```diff
@@ -203,4 +206,35 @@ public static String getHotReloadSupportSource() {
...
+            "import org.drools.ruleunits.api.RuleUnitProvider;\n" +
+            "\n" +
+            "@Dependent\n" +
+            "public class $RULE_UNIT_NAME$Producer {\n" +
+            "\n" +
+            "    @Produces\n" +
+            "    public RuleUnit<$RULE_UNIT_NAME$> produceRuleUnit() {\n" +
+            "        return RuleUnitProvider.get().getRuleUnit(new $RULE_UNIT_NAME$());\n" +
+            "    }\n" +
+            "}\n";
```
Comment (tarilabs) [inline drools-drl-quarkus-extension/drools-drl-quarkus-util-deployment/src/main/java/org/drools/drl/quarkus/util/deployment/DroolsQuarkusResourceUtils.java:233]: help me recall, isn't drools infra using templated resource file for these use-cases? 🤔

**Review by tarilabs — APPROVED** [review]
I am aware I wrote that unit test, but in another context, so suggesting to remove some unneeded comments etc.

Minor comment about templated resource.

File: drools-drl-quarkus-extension/drools-drl-quarkus-util-deployment/src/main/java/org/drools/drl/quarkus/util/deployment/DroolsQuarkusResourceUtils.java (hunk 4 above)
Comment (mariofusco) [reply drools-drl-quarkus-extension/drools-drl-quarkus-util-deployment/src/main/java/org/drools/drl/quarkus/util/deployment/DroolsQuarkusResourceUtils.java:233]: No, we don't use templates here. I could introduce it, but not sure if it worth the pain for just a string replacement.

**Review by evacchi — APPROVED** [review]

**Pull request merged**